src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -183,6 +149,51 @@ public string GraphQLSchema()
             return _cosmosDb.GraphQLSchema;
         }
 
+        private void ParseSchemaGraphQLFieldsForGraphQLType()
+        {
+            string graphqlSchema = GraphQLSchema();
+            DocumentNode root = Utf8GraphQLParser.Parse(graphqlSchema);
```

Comment:
We are already parsing the Cosmos graphql schema in GraphQLSchemaCreator.GenerateCosmosGraphQLObjects() 

Can we please parse this only once and reuse in the other location - based on whichever needs it first?

Reply:
The `` GraphQLSchemaCreator``` is already calling ```GraphQLSchema``` method, since they are parsing different things from the schema file, the only thing that's duplicate is '''DocumentNode root = Utf8GraphQLParser.Parse(graphqlSchema);''', I'm not sure if it makes sense to save the '''DocumentNode root'''

Reply:
Both of them are parsing the same graphql schema file that is input by the user. So, I think saving this root would give us optimal performance.

Reply:
Refactored the duplicate code into a method. :)